frontend/src/pages/main/topup/TopUp.tsx

```diff
@@ -1,11 +1,41 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import {Titles} from "../../../common/constansts/titles";
+import {Box} from "@mui/material";
+import Title from "../../../components/typography/title/Title";
+import TopUpConfirmation from "./confirm/TopUpConfirmation";
+import {IWallet} from "../../../types/wallet";
+import {useAppSelector} from "../../../stores/redux/hooks";
+import {selectLoggedUser} from "../../../stores/redux/loggedUser/loggedUserSlice";
+import {getWalletByUserService} from "../../../services/wallet";
+
+const initialWallet: IWallet = {
+    id: -1,
+    balance: 0,
+    userId: -1
+}
 
 const TopUp: React.FC = () => {
+
+    const [wallet, setWallet] = useState<IWallet>(initialWallet);
```

Comment:
What is the advantage of getting wallet here and not in TopUpConfirmation? Same questions for transactions. Why do you get wallet id in Transcations components and not in TableTranscations

Reply:
in Transcations i faced with loop requests to the server when i tried to get wallet from server, so i moved to the parent component call `getWallet`. now i get `walletId` from redux, so it was removed.